src/Domain/CoordinateMaps/CoordinateMap.tpp

```diff
@@ -25,25 +26,72 @@
 #include "Domain/CoordinateMaps/CoordinateMapHelpers.hpp"
 #include "Domain/CoordinateMaps/TimeDependentHelpers.hpp"
 #include "Domain/FunctionsOfTime/FunctionOfTime.hpp"
-#include "Utilities/Serialization/CharmPupable.hpp"
 #include "Utilities/Gsl.hpp"
 #include "Utilities/MakeArray.hpp"
+#include "Utilities/Serialization/CharmPupable.hpp"
 #include "Utilities/TMPL.hpp"
 #include "Utilities/Tuple.hpp"
+#include "Utilities/TypeTraits/CreateIsCallable.hpp"
+#include "Utilities/TypeTraits/IsA.hpp"
 
 /// \cond
 namespace domain {
+namespace CoordinateMap_detail {
+CREATE_IS_CALLABLE(function_of_time_names)
+CREATE_IS_CALLABLE_V(function_of_time_names)
+
+template <typename T>
+struct map_type {
+  using type = T;
+};
+
+template <typename T>
+struct map_type<std::unique_ptr<T>> {
+  using type = T;
+};
+
+template <typename T>
+using map_type_t = typename map_type<T>::type;
+
+template <typename... Maps, size_t... Is>
+std::unordered_set<std::string> initialize_names(
+    const std::tuple<Maps...>& maps, std::index_sequence<Is...> /*meta*/) {
+  std::unordered_set<std::string> function_of_time_names{};
+  const auto add_names = [&function_of_time_names, &maps](auto index) {
+    const auto& map = std::get<decltype(index)::value>(maps);
+    using TupleMap = std::decay_t<decltype(map)>;
+    constexpr bool map_is_unique_ptr = tt::is_a_v<std::unique_ptr, TupleMap>;
```

Comment:
Why are these sometimes unique_ptrs and sometimes not?  What are they when they're not?

Reply:
The `Composition` coordinate map takes the maps as `unique_ptr<CoordinateMapBase>` while the actual `CoordinateMap` just takes maps as the individual maps (Affine, Rotation, etc...). That's why I needed to handle both here.